### ISSUE TITLE
What is the roadmap for angular-librarian?

### ISSUE THREAD

**Issue by SirDarquan** [gonzofish/angular-librarian#94]
First, I want to give my thanks for creating such a great library that fulfills a sorely missing need in angular development. This helped me add the needed AOT functionality to my library that users were seeking. But I'm not here to plug that. What I'd like to do is open a discussion on where this library is going and the plan to get there.

I completely get that this may be a "temporary" solution for angular library development until the angular-cli team finally makes it a priority but we don't know when that will be. There is also a possibility that this may be an alternate solution for development even after the angular-cli team implements that feature. The problem is we don't know and it's this lack of definition that makes me hesitant to contribute.

During the implementation of my library, I decided early on that I wanted to use a selenium grid service to run my tests against and I got that working pretty well. It helped that my paying job had done this already, so I knew how to make it work. Once I  decided to begin using angular-librarian to add the AOT functionality I thought to myself, "Oh, there's testing built in. That should transfer right over!" It didn't because angular-librarian is extremely opinionated about how testing should work. I had to add several hacks to finally connect to SauceLabs. I know this is functionality others may like to have but my hack is also really specific to my setup.

I have a normal job, a medium to big family (6 kids) and social activities. So my free time is minimal and I need to spend it wisely. The future of this package determines if I make a really robust generic solution, a good enough solution or don't do anything at all as I've got what I need. I know that sounds selfish and I don't intend to be but for me to commit, I need a commitment for the future of the thing I'm contributing to. Does that make sense?

**Comment by gonzofish** [comment]
@SirDarquan absolutely makes sense! No one wants to contribute to a project that's dead. I'm about to start a family myself so I understand the time commitment issues more than ever.

Librarian is in sort of a "haven't heard any complaints" mode. We had one bug, which I hadn't been able to crack (#89, but you did, thanks again) but other than that, since the 1.0 release, I have heard no chatter. I'm not sure if that means whoever is using Librarian is happy or has moved on, but that's why it seems dead.

You can see from the [issues page](https://github.com/gonzofish/angular-librarian/issues) there are some items I'd love to implement, one of which is E2E testing. I have extremely limited knowledge in this arena, which is why I haven't moved on it. But I'd _always_ take a PR on it.

The things that'd be nice to incorporate into Librarian are:
- Monorepo support (#43)
    - Module generator (#42)
- E2E testing (#56, and this issue)
- A bundled examples page (#60)
- Documentation generation (#74, PR #90) - this one will be in the next release

I think these would all take Librarian to the next level. The first two are challenges for me because the monorepo could be a huge paradigm shift on project structure and E2E testing is outside my knowledge at the moment, but again I'm always up for PRs & discussion to move it forward.

The bottom line is that, as long as I am aware its being used, I'll work on the project. So I suppose you shouldn't worry your contributions would be in vain.

**Comment by SirDarquan** [comment]
Cool, I appreciate that. I'll close this and create some issues that address the things I needed to do to hack using a selenium grid providers.
